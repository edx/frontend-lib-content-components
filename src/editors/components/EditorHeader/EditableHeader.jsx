import React from 'react';
import PropTypes from 'prop-types';

import { Icon, Form } from '@edx/paragon';
import { Edit } from '@edx/paragon/icons';

export const EditableHeader = ({
  handleChange,
  updateTitle,
  handleKeyDown,
  inputRef,
  localTitle,
}) => (
  <Form.Group>
    <Form.Control
      autoFocus
      onBlur={updateTitle}
      onChange={handleChange}
      onKeyDown={handleKeyDown}
      placeholder="Title"
      ref={inputRef}
      trailingElement={<Icon src={Edit} />}
      value={localTitle}
    />
  </Form.Group>
);
EditableHeader.defaultProps = {
  inputRef: null,
};
EditableHeader.propTypes = {
  inputRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any }),
  ]),
  handleChange: PropTypes.func.isRequired,
  updateTitle: PropTypes.func.isRequired,
  handleKeyDown: PropTypes.func.isRequired,
  localTitle: PropTypes.string.isRequired,
};

export default EditableHeader;
