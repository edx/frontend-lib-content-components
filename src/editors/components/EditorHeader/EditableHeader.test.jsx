import React from 'react';
import { shallow } from 'enzyme';
import { Icon, Form } from '@edx/paragon';
import { Edit } from '@edx/paragon/icons';
import * as module from './EditableHeader';

describe('EditableHeader', () => {
  const props = {
    handleChange: jest.fn().mockName('args.handleChange'),
    updateTitle: jest.fn().mockName('args.updateTitle'),
    handleKeyDown: jest.fn().mockName('args.handleKeyDown'),
    inputRef: jest.fn().mockName('args.inputRef'),
    localTitle: 'test-title-text',
  };
  let el;
  beforeEach(() => {
    el = shallow(<module.EditableHeader {...props} />);
  });

  describe('snapshot', () => {
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });
    test('displays Edit Icon', () => {
      const formControl = el.find(Form.Control);
      expect(formControl.props().trailingInputElement).toMatchObject(<Icon src={Edit} />);
    });
  });
});
