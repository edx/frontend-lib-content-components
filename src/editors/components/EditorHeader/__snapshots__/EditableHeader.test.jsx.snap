// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableHeader snapshot snapshot 1`] = `
<Form.Group>
  <Form.Control
    autoFocus={true}
    onBlur={[MockFunction args.updateTitle]}
    onChange={[MockFunction args.handleChange]}
    onKeyDown={[MockFunction args.handleKeyDown]}
    placeholder="Title"
    trailingInputElement={
      <Icon
        src={[MockFunction icons.Edit]}
      />
    }
    value="test-title-text"
  />
</Form.Group>
`;
